TestProject/ClientApp/src/components/Posts.js

```diff
@@ -16,7 +17,7 @@ function Posts() {
     return (
         <div>
             {posts.map((post) => (
-                <Post key={post.id} {...post} />
+                <Post key={post.id} post={post} userLoggedIn={userLoggedIn} userProfile={userProfile} />
```

Comment:
Btw `userLoggedIn` and `userProfile` are always passed together. Maybe it make sense to just make `loggedIn` a property of `userProfile`?